CodeStarterCamp_Week4/FitnessCenter.swift

```diff
@@ -0,0 +1,133 @@
+    }
+    
+    func exerciseRoutine() {
+            do {
+                self.member?.name = try writeName()
+                self.targetBodyCondition = try inputBodyCondition()
+            } catch InputError.empty {
+                print("입력되지 않았습니다. 다시 입력해주세요.")
+            } catch InputError.invaildValue {
+                print("없는 번호입니다. 다시 입력해주세요.")
+            } catch InputError.outOfValue {
+                print("값의 갯수가 적거나 큽니다.")
+            } catch {
+                print("알 수 없는 오류입니다. 다시 시도해주세요.")
+            }
+        tryRoutine()
+        }
+    
+    func writeName() throws -> String {
+        print("안녕하세요. \(name)피트니스 센터입니다. 회원님의 이름은 무엇인가요?")
+        guard let inputName = readLine() else { throw InputError.empty }
+        return inputName
+    }
+    
+    func inputBodyCondition() throws -> BodyCondition {
+        print("운동 목표치를 순서대로 알려주세요. [상체근력:  하체근력:  근지구력: ]")
+        print("목표치간 띄어쓰기는 필수입니다. ex) 100 100 100")
+        guard let inputTargetBodyCondition = readLine() else { throw InputError.empty }
+        let targetBodyConditionArray: [Int] = try inputTargetBodyCondition.split(separator: " ").map{
+            guard let number = Int($0) else { throw InputError.invaildValue }
+            return number
+        }
+        
+        if targetBodyConditionArray.count != 3 {
+            throw InputError.outOfValue
+        }
+        
+        let bodyCondition = BodyCondition()
+        
+        return bodyCondition
+    }
+    
+    func tryRoutine() {
+        do {
+            try choiceRoutine()
+        } catch InputError.empty {
+            print("입력되지 않았습니다. 다시 입력해주세요.")
+        } catch InputError.invaildValue {
+            print("없는 번호입니다. 다시 입력해주세요.")
+        } catch InputError.outOfValue {
+            print("값의 갯수가 적거나 큽니다.")
+        } catch {
+            print("알 수 없는 오류입니다. 다시 시도해주세요.")
+        }
+    }
+    
+    func choiceRoutine() throws -> Void {
+        print("몇 번째 루틴으로 운동하시겠어요?")
+        var routineNumber = 1
+        self.listOfRoutine.forEach {
+            print("\(routineNumber). \($0.routineName)")
+            routineNumber += 1
+        }
+        
+        guard let inputRoutine = readLine() else { throw InputError.empty }
+        guard let innerRoutine = Int(inputRoutine) else { throw InputError.invaildValue }
+        
+        switch innerRoutine {
+        case 1..<routineNumber:
+            routineNumber = innerRoutine - 1
+        default: throw InputError.outOfValue
+        }
+        
+        print("몇 세트 반복하시겠어요?")
+        
+        guard let inputSetCount = readLine() else { throw InputError.empty }
+        guard let countNumber = Int(inputSetCount) else { throw InputError.invaildValue }
+        
+        member?.doExercise(for: countNumber, routine: self.listOfRoutine[routineNumber])
+        routineNumber = 1
+        
+        if accomplishCondotion() {
```

Comment:
피로도가 100 을 넘어도, `무한루프` 돈다고 하셨죠?
이 부분에 `guard문` 하나 추가해서 해결할 수 있습니다! 👍🏻 

```suggestion
        routineNumber = 1
        
        // 피로도 100이면 프로그램이 자연스럽게 종료되도록 하기
        guard member?.bodyCondition.fatigue ?? 0 < 100 else {
            print("[알림] 운동을 할 회원이 없으므로, 프로그램이 종료됩니다.")
            return
        }
        
        if accomplishCondition() {
```

Reply:
ㅇ0ㅇ!! 이런 방법이 있었군요...!